src/engine/enginebuffer.cpp

```diff
@@ -443,10 +443,15 @@ void EngineBuffer::requestSyncMode(SyncMode mode) {
     }
 }
 
-void EngineBuffer::clearScale() {
+void EngineBuffer::clearScale(EngineBufferScale* oldScale) {
```

Comment:
This code is hard to understand from the function name and from the control flow. 
Does it make sense to change it to a 
setScale(EngineBufferScale* newScale)
Function and move some of the calling code into it? 
That should be clarify things.